csrc/scheduler/matmul_utils.cpp

```diff
@@ -294,6 +315,98 @@ std::string isMatmulFusionDefinitionSupported(
   return "";
 }
 
+// Assume that tens has a contiguous dimension, and that we will load rows of
+// that dimension, without merging with another dimension first. Then determine
+// the maximum vectorization that can be used.
+//
+// If the argument has no contiguous dimensions, then a vectorization width of 1
+// is returned.
+//
+// These rows can start at any multiple of the non-contiguous strides, so we
+// seek the largest power of 2 that divides all those other dimensions (capped
+// to 16) as well as the data pointer.
+int64_t maxRowVectorization(const at::Tensor& tens) {
+  const int64_t data_ptr_int =
+      static_cast<int64_t>(reinterpret_cast<std::uintptr_t>(tens.data_ptr()));
+  int64_t vec_size = scheduler_utils::maxVectorizationWidth(data_ptr_int);
+  vec_size = std::min(vec_size, (int64_t)(16l));
+  vec_size /= (int64_t)tens.element_size();
+  vec_size = std::max(vec_size, 1l);
+  if (vec_size == 1l) {
+    return vec_size;
+  }
+
+  std::vector<int64_t> strides = tens.strides().vec();
+  std::sort(strides.begin(), strides.end());
+  NVF_ERROR(!strides.empty());
+  if (strides.front() != 1l) {
+    // Tensor is discontiguous
+    return 1l;
+  }
+  strides.erase(strides.begin());
+  for (auto stride_i : strides) {
+    if (vec_size == 1l) {
+      break;
+    }
+    vec_size =
+        std::min(vec_size, scheduler_utils::maxVectorizationWidth(stride_i));
+  }
+
+  return vec_size;
+}
+
+MatmulParams::SupportedVectorization getSupportedVectorization(
```

Comment:
Could you split this PR into two?
1. Adding SupportedVectorization to heuristics and modify matmul scheduler to respect this field in heuristics
2. Computing SupportedVectorization